fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -312,8 +317,18 @@ private static void createOutputFragment(PlanFragment inputFragment, ExecPlan ex
         execPlan.getFragments().add(exchangeFragment);
     }
 
-    static boolean useQueryCache(ExecPlan execPlan) {
-        if (ConnectContext.get() == null || !ConnectContext.get().getSessionVariable().isEnableQueryCache()) {
+    private static boolean enableComputeNode(ExecPlan execPlan) {
+        if (execPlan.getConnectContext().getSessionVariable().getUseComputeNodes() > 0) {
```

Comment:
Maybe needn't this condition, because, non-positive `useComputeNodes` indicates there isn't limitation of the maximum number of compute nodes to use.

```java
    private ImmutableMap<Long, ComputeNode> buildComputeNodeInfo() {
        ImmutableMap<Long, ComputeNode> idToComputeNode
                = ImmutableMap.copyOf(GlobalStateMgr.getCurrentSystemInfo().getIdComputeNode());
        int useComputeNodeNumber = connectContext.getSessionVariable().getUseComputeNodes();
        if (useComputeNodeNumber <= 0 || useComputeNodeNumber >= idToComputeNode.size()) {
            return idToComputeNode;
        } else {
            Map<Long, ComputeNode> computeNodes = new HashMap<>();
            for (int i = 0; i < useComputeNodeNumber; i++) {
                ComputeNode computeNode = SimpleScheduler.getComputeNode(idToComputeNode);
                if (computeNode == null) {
                    continue;
                }
                computeNodes.put(computeNode.getId(), computeNode);
            }
            return ImmutableMap.copyOf(computeNodes);
        }
    }
```

Reply:
ok, i get it

Reply:
Maybe `buildComputeNodeInfo()` also need modify? 
```diff
- if (useComputeNodeNumber < 0 || useComputeNodeNumber >= idToComputeNode.size()) {
+ if (useComputeNodeNumber <= 0 || useComputeNodeNumber >= idToComputeNode.size()) {
```

Reply:
emm, I think don't need modify Coordinator, `GATHER` plan is right for normal node mode